tests/zfs-tests/tests/perf/regression/random_reads.ksh

```diff
@@ -31,6 +31,15 @@
 
 function cleanup
 {
+	# kill fio and iostat
+	$PKILL ${FIO##*/}
+	$PKILL ${IOSTAT##*/}
+
+	# fio and iostat can be stuck in a sleep/pause call
+	while pgrep "(${FIO##*/})|(${IOSTAT##*/})"; do
+		sleep 5
+	done
```

Comment:
Using `log_must_busy` below for the destroy probably a simpler option.  In the case where it's really busy you'll get a reasonable backoff and retry, and in the case where something else is wrong it will immediately fail.